helper-cli/src/main/kotlin/commands/SetDependencyRepresentationCommand.kt

```diff
@@ -0,0 +1,157 @@
+import org.ossreviewtoolkit.model.AnalyzerResult
+import org.ossreviewtoolkit.model.OrtResult
+import org.ossreviewtoolkit.model.ProjectAnalyzerResult
+import org.ossreviewtoolkit.model.readValue
+import org.ossreviewtoolkit.model.utils.DependencyGraphConverter
+import org.ossreviewtoolkit.model.writeValue
+import org.ossreviewtoolkit.utils.expandTilde
+
+/**
+ * A command to convert the dependency representation in an ORT result file between the different supported formats.
+ *
+ * The input file can contain the serialized form of either a full [OrtResult], an [AnalyzerResult], or a single
+ * [ProjectAnalyzerResult]. If one of the supported results is found, it is converted to the specified target format.
+ * If this target format is already used, then this command does nothing.
+ */
+class SetDependencyRepresentationCommand : CliktCommand(
+    help = "Convert the dependency representation of an ORT result to a specific target format."
+) {
+    /**
+     * An enumeration class representing the different supported formats to represent dependencies in an ORT result.
+     * It is used to define the target format that should be set by this command.
+     */
+    enum class TargetFormat {
+        /** The dependency graph format. */
+        GRAPH {
+            override fun convert(result: AnalyzerResult): AnalyzerResult =
+                DependencyGraphConverter.convert(result)
+        };
+
+        /**
+         * Convert the given [result] to the target format defined by this literal.
+         */
+        abstract fun convert(result: AnalyzerResult): AnalyzerResult
+    }
+
+    private val ortFile by option(
+        "--ort-file", "-i",
+        help = "The ORT result file to read as input. This can be the serialized form of an OrtResult, an " +
+                "AnalyzerResult, or a ProjectAnalyzerResult. NOTE: If no output file is specified, this file is " +
+                "overwritten."
+    ).convert { it.expandTilde() }
+        .file(mustExist = true, canBeFile = true, canBeDir = false, mustBeWritable = false, mustBeReadable = true)
+        .convert { it.absoluteFile.normalize() }
+        .required()
+
+    private val outputFile by option(
+        "--output-file", "-o",
+        help = "The file in which to write the result of the conversion. If missing, the input file is overwritten."
+    ).convert { it.expandTilde() }
+        .file(mustExist = false, canBeFile = true, canBeDir = false, mustBeWritable = false, mustBeReadable = false)
+        .convert { it.absoluteFile.normalize() }
+
+    private val targetFormat by option(
+        "--format", "-f",
+        help = "The target format for the conversion."
+    ).enum<TargetFormat>().default(TargetFormat.GRAPH)
+
+    override fun run() {
+        val converters = listOf(::convertOrtResult, ::convertAnalyzerResult, ::convertProjectAnalyzerResult)
+        converters.asSequence().mapNotNull { it() }.firstOrNull()?.let { writeResult(it) }
```

Comment:
If we view `converters` as a sequence, we could as well right away use `sequenceOf()` instead of `listOf()`.